company-template.el

```diff
@@ -65,15 +66,27 @@
            (> pos (point))
            (< pos minimum)
            (setq minimum pos)))
-    (push-mark)
-    (goto-char minimum)
-    (company-template-remove-field (company-template-field-at start))))
+    (let ((start-ovl (company-template-field-at start)))
+      (if delete-template
```

Comment:
> Instead of having two different behaviors inside a function, I think I'd rather the first part of this function was extracted to a utility one, and both company-template-forward-field and company-template-forward-field-and-delete called it

Sounds good. I'll update the patch with your suggestions